prow/cmd/artifact-uploader/main.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/test-infra/prow/kube"
+
+	"k8s.io/test-infra/prow/artifact-uploader"
+	"k8s.io/test-infra/prow/gcsupload"
+	"k8s.io/test-infra/prow/logrusutil"
+	"k8s.io/test-infra/prow/pod-utils/options"
+)
+
+// NewOptions returns an empty Options with no nil fields
+func NewOptions() *Options {
+	return &Options{
+		Options: gcsupload.NewOptions(),
+	}
+}
+
+type Options struct {
+	// NumWorkers determines the number of workers that consume
+	// the controller's work queue
+	NumWorkers int `json:"num_workers"`
+	// ProwJobNamespace is the namespace in the cluster that holds
+	// ProwJob objects
+	ProwJobNamespace string `json:"'prow_job_namespace'"`
+
+	*gcsupload.Options
+
+	clusterConfig *rest.Config
+}
+
+// Validate ensures that the set of options are
+// self-consistent and valid
+func (o *Options) Validate() error {
+	if o.NumWorkers == 0 {
+		return errors.New("number of workers cannot be zero")
+	}
+
+	if o.ProwJobNamespace == "" {
+		return errors.New("namespace containing ProwJobs not configured")
+	}
+
+	return o.Options.Validate()
+}
+
+const (
+	// JSONConfigEnvVar is the environment variable that
+	// utilities expect to find a full JSON configuration
+	// in when run.
+	JSONConfigEnvVar = "ARTIFACTUPLOAD_OPTIONS"
+)
+
+// ConfigVar exposes the environment variable used
+// to store serialized configuration
+func (o *Options) ConfigVar() string {
+	return JSONConfigEnvVar
+}
+
+// LoadConfig loads options from serialized config
+func (o *Options) LoadConfig(config string) error {
+	return json.Unmarshal([]byte(config), o)
+}
+
+// BindOptions binds flags to options
+func (o *Options) BindOptions(flags *flag.FlagSet) {
+	flags.IntVar(&o.NumWorkers, "num-workers", 25, "Number of threads to use for processing updates.")
+	flags.StringVar(&o.ProwJobNamespace, "prow-job-ns", "", "Namespace containing ProwJobs.")
```

Comment:
We already store the prowjob namespace in the prow config, although it's still static.

Reply:
Loading the entire config seemed like overkill for one field.